.github/workflows/ci.yml

```diff
@@ -224,7 +226,7 @@ jobs:
     steps:
     - run: exit 1
       name: At least one check failed or was cancelled
-      if: ${{ !(success()) }}
+      if: ${{ contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled') }}
```

Comment:
I don't understand the reason for the change here, could you explain?

Reply:
At the step level, success is checking that each previous step succeeded, not that the required jobs succeeded, see e.g. https://github.com/py-why/EconML/actions/runs/4461230788/jobs/7835479732